src/runner/patches.js

```diff
@@ -86,3 +86,8 @@ Math.pow = new Proxy(
     args.every((arg) => arg instanceof Decimal || arg instanceof Big)
   )
 );
+
+Decimal.round = new Proxy(
+  Decimal.round,
+  createNaryHandler(roundImpl, roundRefiner)
```

Comment:
So this doesn't quite work as expected — `createNaryHandler` dispatches to the target or to the handler based on whether none or all of the arguments are decimals, respectively. This means that when we call `Decimal.round(1.5, { roundingMode: "up", maximumFractionDigits: 0 })` we are actually calling decimal.js's `Decimal.round()`, so the line in the refiner that throws in this case is never reached.

We could rewrite `createNaryHandler` to call the refiner before we dispatch to the target, but that would break `Math.pow` unless we write a more complicated refiner for it. That's not impossible, but I thought it'd be better to replace `Decimal.round` as the target here (since the proposal's `Decimal.round` should never call into decimal.js's `Decimal.round`, anyway) with a function that throws, and use `createUnaryHandler` instead. This is what I've done in #14, but we could also take this alternate route.

Reply:
Yes, the draft spec specifies that when called with a  non-`Decimal` type, `round` should throw a `TypeError`. 

I am open to the idea that we will change the spec based on feedback (and I could certainly think about whether it would be desireable to have `Decimal.round` implicity coerce the type of the value as a nicer instantiation and a way to deal with https://github.com/tc39/proposal-decimal/issues/40), I think right now it's good to focus on implementing what is spec'd and possibly opening issues to discuss change ideas we might have.